libcontainer/cgroups/fs/apply_raw.go

```
@@ -134,6 +145,16 @@ func (m *Manager) Apply(pid int) (err error) {
 	defer m.mu.Unlock()
 	paths := make(map[string]string)
 	for _, sys := range subsystems {
+		path, err := cgroups.GetOwnCgroupPath(sys.Name())
+		if err != nil {
+			// Ignore paths we couldn't resolve.
+			continue
+		}
+		if !m.CanWrite(path) {
```

The error should actually be on `Set`, I was a bit confused when reviewing this PR (since the access logic was duplicated). Using the "list of enabled cgroups" we discussed below, we just need to make sure that the disabled cgroups aren't necessary for setting up the container as requested (otherwise we're silently ignoring resource restrictions, which is bad). My point is that we should never be silently ignoring a resource restriction.

The `Apply` stuff with lazy cgroup joining isn't necessary for this PR (though you can do it separately against runC master if you like). Also, that code will complicate quite a few things (you need to deal with `Set` that has a different config to the one specified at start time).